### PR TITLE
Create repositories, branches, commits, and merges using time on DB

### PR DESCRIPTION
**Pull request by arielshaqed**
Using system time [causes weirdness](https://youtu.be/-w0WPkB3XJ4) when there are multiple
lakefs daemons.

### CODE REVIEW

**Comment by arielshaqed** [comment]
With this change:
```
ariels@ariels:~/Dev/lakeFS$ git grep -F -w Now catalog/ | fgrep -v _test.go
catalog/cataloger.go:                           Timestamp:          time.Now(),
catalog/db_batch_entry_read.go:                                 startTime: time.Now(),
```
and these 2 entries are for dedupe batching and never touch the database.

File: catalog/cataloger_commit_test.go [hunk 1]
```diff
@@ -97,6 +95,9 @@ func TestCataloger_Commit(t *testing.T) {
 				t.Errorf("Commit() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
+			if got != nil {
+				got.CreationDate = got.CreationDate.Truncate(time.Minute)
```
Comment (nopcoder) [inline catalog/cataloger_commit_test.go:99]: This one will fail in some cases - skip the check my assign the `want` value or time default value for both.
The previous version we used the Truncate trick to reset the 'monotonic clock' value.

File: catalog/cataloger_get_commit_test.go [hunk 2]
```diff
@@ -97,6 +93,9 @@ func TestCataloger_GetCommit(t *testing.T) {
 				t.Errorf("GetCommit() error = %s, wantErr %t", err, tt.wantErr)
 				return
 			}
+			if got != nil {
+				got.CreationDate = got.CreationDate.Truncate(time.Minute)
```
Comment (nopcoder) [inline catalog/cataloger_get_commit_test.go:97]: same as before - don't want to play sync my clock with the database - prefer to skip the creation time or just verify once it is ~ the same time and not truncate and have the test fail once when we cross the minute.

File: docs/assets/js/swagger.yml [hunk 3]
```diff
@@ -225,10 +225,11 @@ definitions:
   setup:
     type: object
     properties:
-      display_name:
+      username:
+        description: an identifier for the user (e.g. jane.doe)
         type: string
     required:
-      - display_name
+      - username
```
Comment (nopcoder) [inline docs/assets/js/swagger.yml:232]: this one looks like it is part of another change that was reverted but a copy left in the docs.

File: catalog/cataloger_commit_test.go (hunk 1 above)
Comment (arielshaqed) [reply catalog/cataloger_commit_test.go]: Replaced with code that tests roughly what we want: that times are close to one another.  Yes, it's flaky when clocks drift apart, but OTOH we're running a newly-started Postgres container -- even on MacOS the time skew will be minimal.

File: catalog/cataloger_get_commit_test.go (hunk 2 above)
Comment (arielshaqed) [reply catalog/cataloger_get_commit_test.go]: Yup, as above.

**Review by arielshaqed — COMMENTED** [review]
Thanks; PTAL!

**Review by nopcoder — APPROVED** [review]

**Comment by arielshaqed** [comment]
Thanks!
Pulling...

**Comment by nopcoder** [comment]
🙏  it will not break

**Comment by arielshaqed** [comment]
> pray it will not break

Noooooo!!  Don't jinx it!

**Pull request merged**